crates/node/src/storage/rocksdb.rs

```diff
@@ -688,12 +734,148 @@ impl RocksDB {
             old_and_new_diff_key(key, height)?.1
         };
 
-        self.0
+        self.inner
             .get_cf(rollback_cf, key)
             .map_err(|e| Error::DBError(e.into_string()))
     }
 }
 
+pub struct DbSnapshot<'a>(pub rocksdb::Snapshot<'a>);
+
+impl<'a> DbSnapshot<'a> {
+    /// Write a snapshot of the database out to file. The last line
+    /// of the file contains metadata about how to break the file into
+    /// chunks.
+    pub fn write_to_file(
+        &self,
+        cfs: [(&'static str, &'a ColumnFamily); 6],
+        path: &Path,
+    ) -> std::io::Result<()> {
+        let file = File::create(path)?;
+        let mut buf = BufWriter::new(file);
+        let mut chunker = Chunker::new(MAX_CHUNK_SIZE);
+        for (cf_name, cf) in cfs {
+            let read_opts = make_iter_read_opts(None);
+            let iter =
+                self.0.iterator_cf_opt(cf, read_opts, IteratorMode::Start);
+
+            for (key, raw_val, _gas) in PersistentPrefixIterator(
+                PrefixIterator::new(iter, String::default()),
+                // Empty string to prevent prefix stripping, the prefix is
+                // already in the enclosed iterator
+            ) {
+                let val = HEXLOWER.encode(&raw_val);
+                let bytes = format!("{cf_name}:{key}={val}\n");
+                chunker.add_line(&bytes);
+                buf.write_all(bytes.as_bytes())?;
+            }
+            buf.flush()?;
+        }
+        let chunks = chunker.finalize();
+        let val = HEXLOWER.encode(&chunks.serialize_to_vec());
```

Comment:
I think base64 might be better for this usecase, as it is slightly more efficient than hex encoding in terms of space usage. ideally, snapshots would be stored in a binary format, anyway, to save even more space

Reply:
I think the snapshot format can be greatly improved. But I wanted to get a working version of state sync first since we can iterate on the snapshot format easily later.

Reply:
But I can change it to base64 for now